src/named.rs

```diff
@@ -391,3 +402,89 @@ impl<'a, 'b> NamedTempFileOptions<'a, 'b> {
 
     }
 }
+
+/// A path to a named temporary file without an open file handle.
+///
+/// This is useful when the temporary file needs to be used by a child process,
+/// for example.
+///
+/// When dropped, the temporary file is deleted.
+pub struct TempPath(PathBuf);
+
+impl TempPath {
+    /// Close and remove the temporary file.
+    ///
+    /// Use this if you want to detect errors in deleting the file.
+    pub fn close(self) -> io::Result<()> {
+        fs::remove_file(&self.0)
+    }
+
+    /// Persist the temporary file at the target path.
+    ///
+    /// If a file exists at the target path, persist will atomically replace it.
+    /// Upon failure, `self` is returned in the error result.
+    ///
+    /// Note: Temporary files cannot be persisted across filesystems.
+    ///
+    /// *SECURITY WARNING:* Only use this method if you're positive that a
+    /// temporary file cleaner won't have deleted your file. Otherwise, you
+    /// might end up persisting an attacker controlled file.
+    pub fn persist<P: AsRef<Path>>(self, new_path: P) -> Result<(), (TempPath, io::Error)> {
```

Comment:
This error type is the only thing I'm not 100% happy about. I can't use the existing `PersistError` because it isn't general enough. Any ideas? Or is this fine?

Reply:
An alternative is to just provide a method for reopening the file (we can then persist from `NamedTempFile`). However, this would be slightly less efficient. Regardless, I'd like such a method anyways (e.g., `open()`).

We could also just create a new error type. I'll leave that up to you. However, I *wouldn't* just use a tuple like this; `PersistError` has some convenience conversion functions that are kind of useful.